src/sentry/static/sentry/app/components/forms/apiForm.tsx

```diff
@@ -40,14 +41,32 @@ export default class ApiForm extends Form<Props> {
     this.api.clear();
   }
 
+  getEnabledData() {
+    // Return a hash of data from non-disabled fields.
+    const data = {...this.state.data}; // copy to avoid mutating original
+    if (Array.isArray(this.props.children)) {
+      this.props.children.forEach(child => {
+        // Bit of a hack: loop over components named *Field to filter out disabled ones.
+        if (child.type && child.type.name && child.props && child.key) {
+          if (/Field$/.test(child.type.name)) {
+            if (child.props.disabled) {
```

Comment:
```suggestion
        const childTypeName = child.type && child.type.name || '';
        if (/Field$/.test(childTypeName) && child.props?.disabled && child.key) {
```

Reply:
Do we not care about `child.props` and `child.key`? This is my first time working against a typechecker in JavaScript, I need to tune my internal dial here. :)

Reply:
I think they are expected to be there but I followed your lead, just merged the nested `if` blocks into a single condition joined by `&&`, which is equivalent?

Reply:
My bad, I misread.

Reply:
TIL [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) 👍